int32 id
string message
int32 position
int32 storage
int32[] slot
